sway/server.c

```diff
@@ -32,16 +42,29 @@ static void server_ready(struct wl_listener *listener, void *data) {
 	}
 }
 
-bool server_init(struct sway_server *server) {
+bool server_init(struct sway_server *server, bool headless) {
 	wlr_log(L_DEBUG, "Initializing Wayland server");
 
 	server->wl_display = wl_display_create();
 	server->wl_event_loop = wl_display_get_event_loop(server->wl_display);
-	server->backend = wlr_backend_autocreate(server->wl_display);
 
-	server->renderer = wlr_gles2_renderer_create(server->backend);
+	wl_list_init(&server->subbackends);
+
+	if (headless) {
+		server->backend = wlr_multi_backend_create(server->wl_display);
+		struct sway_subbackend *subbackend =
+			sway_subbackend_create(SWAY_SUBBACKEND_HEADLESS, "headless");
+		sway_server_add_subbackend(server, subbackend);
+		sway_subbackend_add_output(server, subbackend, "headless");
+	} else {
+		// TODO add whatever this function creates to the subbackends
```

Comment:
Will need a test here eventually to see if autocreate gave us a multibackend that we can just reuse or if we need to create a multibackend ourselves.

Reply:
Maybe wlroots should provide some helper functions for merging multibackends together?